pandas/io/s3.py

```diff
@@ -1,11 +1,17 @@
+from typing import Optional, Tuple, TYPE_CHECKING
+from pandas._typing import FilePathOrBuffer
+
 """ s3 support for remote file interactivity """
 from urllib.parse import urlparse as parse_url
 
 from pandas.compat._optional import import_optional_dependency
 
-s3fs = import_optional_dependency(
-    "s3fs", extra="The s3fs package is required to handle s3 files."
-)
+if TYPE_CHECKING:
```

Comment:
Generally trying to avoid TYPE_CHECKING though understood use here. I think better to use a higher level IO annotation if applicable or leave the optional dependency untyped